### PR TITLE
Added `StopOnFailure` Parameter to `RepeatAttribute`

### LINKED ISSUE #4861: Always repeat Test x times
Currently only repeat until success `RetryAttribute` and repeat until failure `RepeatAttribute` are possible.
There should be a way to repeat a test always x times, for stress testing a system for example.

This would probably need the refactor mentioned in the `RepeatAttribute` [here](https://github.com/nunit/nunit/blob/a39e8297b4e9ad279679430021af7a83f3091d59/src/NUnitFramework/framework/Attributes/RepeatAttribute.cs#L3)

### PR DESCRIPTION
**Pull request by cfuerbachersparks**
Changes to always repeat a test x times.
Fixes #4861.

### CODE REVIEW

**Comment by cfuerbachersparks** [comment]
@dotnet-policy-service agree company="Sparks Solutions GmbH"

File: src/NUnitFramework/testdata/TestFixtureSourceData.cs [hunk 1]
```diff
@@ -72,10 +72,7 @@ public StaticProperty_SameClass(string arg, string expected) : base(arg, expecte
         {
         }
 
-        public static object[] StaticProperty
-        {
-            get { return new object[] { new object[] { "StaticPropertyInClass" } }; }
-        }
+        public static object[] StaticProperty => new object[] { new object[] { "StaticPropertyInClass" } };
```
Comment (manfred-brands) [inline src/NUnitFramework/testdata/TestFixtureSourceData.cs:75]: Please revert this change as it is not related to the changes for the `RepeatAttribute`

**Review by manfred-brands — CHANGES_REQUESTED** [review]
Thanks @cfuerbachersparks. The actual change to functionality and tests looks good.
I have 2 requests for changes, see individual comments.

File: src/NUnitFramework/framework/Attributes/RepeatAttribute.cs [hunk 2]
```diff
@@ -21,14 +22,17 @@ namespace NUnit.Framework
...
         private readonly int _count;
+        private readonly bool _stopOnFailure;
 
         /// <summary>
         /// Construct a RepeatAttribute
         /// </summary>
         /// <param name="count">The number of times to run the test</param>
-        public RepeatAttribute(int count) : base(count)
+        /// <param name="stopOnFailure">Whether to stop when a test is not successful or not</param>
+        public RepeatAttribute(int count, bool stopOnFailure = true) : base(count)
```
[the remark this answers is not in the document]
Comment (cfuerbachersparks) [reply src/NUnitFramework/framework/Attributes/RepeatAttribute.cs]: made `StopOnFailure` a property, restored old ctor and created a new one with non-optional parameter

File: src/NUnitFramework/testdata/TestFixtureSourceData.cs (hunk 1 above)
Comment (cfuerbachersparks) [reply src/NUnitFramework/testdata/TestFixtureSourceData.cs]: reverted changes

**Comment by cfuerbachersparks** [comment]
> Thanks @cfuerbachersparks.
> 
> One more item, can you add a unit test which uses the attribute and sets the `StopOnError` property, one each for `false` and `true`

added two unit tests

**Review by manfred-brands — APPROVED** [review]
Sorry for the delay. Looks good now.
